### DISCUSSION TITLE
Plotly Modbar missing

### DISCUSSION THREAD

**Discussion by thomasmore72** [datapane/datapane#206] [Help] [answered]
Hi all,

When I upload my Plotly graph to Datapane I loose the Modbar tools I added and it's rotated vertically.
here are the tools using fig.show():
![image](https://user-images.githubusercontent.com/77403920/172721888-6ee82984-9984-4bed-b6b8-3e5b444dcad2.png)
and here are the tools once uploaded to Datapane:
![image](https://user-images.githubusercontent.com/77403920/172722179-3868b066-a0c7-4c90-8d54-5fd236519e77.png)
Does anyone have any ideas?

**Comment by shahinrostami** [comment]
Hi @thomasmore72! 

Thanks for sharing this I'm going to take a look at it now. Can you share a minimal example that presents the issue?

**Reply by thomasmore72** [reply]
Not entirely sure what you want but here's the entire script: (its not sensitive as its just a test script, might be a bit messy
[Test script.txt](https://github.com/datapane/datapane/files/8870348/Test.script.txt)
)

**Comment by shahinrostami** [comment, marked as answer]
Thank you @thomasmore72! 

I took a closer look and found that you can configure the mode bar with `update_layout`, e.g.:

```
fig.update_layout(modebar_add=config['modeBarButtonsToAdd'])

import datapane as dp
dp.Report(dp.Plot(fig)).upload(name="My example reportv5", open=True, publicly_visible=True)
```

Check it out [here](https://datapane.com/reports/XkndWB3/my-example-reportv5/), besides the orientation is that what we're after?

The orientation of the mode bar looks like a bug - I'll convert this to an issue!

Please let us know if there's anything else in the meantime!

**Reply by thomasmore72** [reply]
How do I get you a pay rise? haha thanks I really appreciate that
